nntrainer/tensor/swap_device.h

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2022 Jiho Chu <jiho.chu@samsung.com>
+ *
+ * @file   swap_device.cpp
+ * @date   01 July 2022
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jiho Chu <jiho.chu@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  Swap device class
+ *
+ */
+
+#ifndef __SWAP_DEVICE_H__
+#define __SWAP_DEVICE_H__
+
+#include <fcntl.h>
+#include <map>
+#include <memory>
+#include <string>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <system_error>
+#include <unistd.h>
+#include <utility>
+
+
+/* Uncomment this to use mmap for swap data */
+//#define USE_MMAP
+
+namespace nntrainer {
+
+/**
+ * @class   SwapDevice
+ * @brief   A device used to storing data with long access time
+ */
+class SwapDevice {
+public:
+  /**
+   * @brief swap device default path
+   *
+   */
+  const std::string swap_device_path = "/tmp/";
```

Comment:
It might be better set this using property from user in later PR

Reply:
OK. I'll add other property.